### PR TITLE
Alerting: Preserve new-lines from custom email templates in rendered email

### LINKED ISSUE #52310: New lines are remove from Alerting templates when used with e-mail contact point.
**What happened**:
If you use the default [example template](https://grafana.com/docs/grafana/latest/alerting/contact-points/message-templating/example-template/) for an e-mail contact point, it removes all the new lines, making the templating feature for e-mail not very friendly. I tried adding extra new lines (`\n`) but no luck. Do we have a secret trick?

**What you expected to happen**:
That new lines are kept when using the template for e-mail.

**How to reproduce it (as minimally and precisely as possible)**:
- Create the [example templates](https://grafana.com/docs/grafana/latest/alerting/contact-points/message-templating/example-template/)
- Create an e-mail contact point and set the message to `{{ template "mymessage" . }}`

**Environment**:
- Grafana Cloud / GE: 9.0.*

### PR DESCRIPTION
**Pull request by alexweav**
**What this PR does / why we need it**:

Right now, any customized email messages from alerting display with the newlines destroyed:

image

This is because the newlines aren't changed to `<br>`'s, nor is any stying applied to that block aside from the parent styling.

**Which issue(s) this PR fixes**:

This PR adds some simple styling to this message block (if it's present) that preserves newlines transparently. Now, newlines from the Alerting template show up in the actual email by default, completely as the user would expect.

image

rel https://github.com/grafana/grafana/issues/42946

Fixes #52310

**Special notes for your reviewer**:

There are a lot of unrelated changes in this PR. Essentially, several changes have gone into the code-generated email templates that were implemented by hand. So, if I run the template generator, all these changes get un-done (!!!)

I dug through the git history and found every such PR, and hand-recreated them in the generator side. Ran the generator and now everything is back in alignment!

There are still a few minor changes that the generator added, which are all ones that we actually do want to preserve or have no effect. These include:
- Whitespace produced by the generator
- `border` -> `border-width` changes
- Some inherited class names (whose contents are inlined, so this doesn't affect anything)

### CODE REVIEW

File: emails/templates/ng_alert_notification.html [hunk 1]
```diff
@@ -101,7 +116,7 @@
 [[ end ]]
 
 [[ if gt (len .Message) 0 ]]
- [[ .Message ]]
+  <span style="white-space: pre-line;">[[ .Message ]]</span>
```
Comment (alexweav) [inline emails/templates/ng_alert_notification.html:119]: This is the only real change that relates to this PR. The rest of the changes are recreations of past PRs, in the generator.

**Review by gotjosh — APPROVED** [review]
LGTM, but how do we make sure this stays up to day _always_?

**Review by JacobsonMT — APPROVED** [review]
LGTM. I think I agree with @joeblubaugh on left aligning the lines, but either way it's a good improvement.

**Comment by alexweav** [comment]
After trying it, definitely agree with @joeblubaugh and @JacobsonMT, left aligning was a great suggestion and it looks even better. Thanks!

**Comment by alexweav** [comment]
Holding off on the CI step or anything of that nature for now, to keep this PR targeted.

**Comment by basvdl** [comment]
Is my understanding correct that this will be part of the 9.0.4 release?

**Comment by MrDibbley** [comment]
Running 9.0.4 (ng_alert_notification.html has above changes) and emails still wrapping without new lines? does something need configuring in the template to enable this ?

Thanks

**Comment by anwu** [comment]
Seconding @MrDibbley here.
I'm on 9.0.5 and still am having issues with the newlines not being preserved.
@alexweav I see in your screenshot that your version is 9.1.0-pre. Is it going to be in 9.1.0 and the notes in 9.0.4 are incorrect?

**Comment by alexweav** [comment]
@MrDibbley @anwu Sorry to hear there are still issues! Firstly, this change definitely made it into 9.0.4 - although the screenshots above say 9.1.0, we did also [backport](https://github.com/grafana/grafana/issues/49256) this to the 9.0 releases.

Our email templating system is fairly confusing right now, we're working on improving the whole template experience as a whole. There are two layers of templating happening and this change only affects the second layer. If no newlines made it to this file in the first place, this PR won't appear to change anything. This is the first thing to check - is your original alert template sending newlines?

FWIW, I installed a 9.0 version (happened to be 9.0.7, but this has not changed since 9.0.4 so it should be the same). I used the following test template:
```
{{ len .Alerts.Firing }} firing:

{{-  range .Alerts.Firing }}
[{{.Status}}] {{ .Labels.alertname }}:
Labels:
  {{- range .Labels.SortedPairs }}
  - {{ .Name }}: {{ .Value }}
  {{- end }}
{{ end }}
```
It showed the newlines (and left-align) as expected:
2022-08-16-113707_420x270_scrot

The other thing to check, is that this `ng_alert_notification` file is only used by Unified Alerting. If you are using Legacy alerting, it uses yet another different templating system and thus you might not see this change take effect. The solution for legacy might not be the same as this.

Hopefully this helps you both! If you are still having problems, please feel free to open an issue and we can take a deeper look.

**Comment by iStep2Step** [comment]
Is there a chance that this is going to be fixed for older versions too?

**Comment by yzaltz** [comment]
Using latest version 9.3.2 - and the problem exist for me 

My template:
{{ with $values }}
  {{  range $k, $v := . }}
  {{ if $v.Labels.action }}
     - Action: {{ $v.Labels.action }}
   {{ end }}
   {{ if $v.Labels.service }}
      - Service: {{ $v.Labels.service }}
    {{ end }}
   Value: {{ $v }}
  {{ end }}
{{ end }}

The same notification is sent to Outlook and Telegram - 
The Outlook email output doesn't include the newline while the Telegram does (see below) is this bug really fixed?
Outlook:
outlook
Telegram:
telegram

**Comment by gillesdemey** [comment]
Outlook does not support `white-space: pre-line` introduced [in the PR](https://github.com/grafana/grafana/pull/52470/files#diff-09633261e0edd9dc2520c1022999c518a583c78e3466de22cd4aa11f623377fdR119). We did not know that Outlook and some other mail clients do not support this CSS property.

Grafana v9.4 will be released with updated email templates and those will parse newlines [in the template renderer instead](https://github.com/grafana/grafana/pull/57751/files#diff-bee1ccc3537bfe55eb9815f39a08ae8254301d6c14c7bb9447c537010b9c1af5R27-R33).

**Pull request merged**